### PR TITLE
Modernize code

### PR DESCRIPTION
**Pull request by RyanZim**
Note that this PR is against the `develop` branch.

I don't know if we should upgrade the JS API to either switch to Promises only, or support both Promises and callbacks. Thoughts?

### CODE REVIEW

**Comment by RyanZim** [comment]
Tests fail on Windows, investigating.

**Comment by RyanZim** [comment]
This is strange; I added a `console.log` for debugging, and the build passed: https://github.com/giakki/uncss/commits/appveyor

I force-pushed here, and it failed except for the Node 6 job on the PR build. Can somebody on Windows pull this in and test locally?

File: src/uncss.js [hunk 1]
```diff
@@ -206,23 +205,30 @@ function init(files, options, callback) {
...
         }
         return getStylesheets(opts.files, opts, pages)
-          .spread(getCSS)
-          .spread(processWithTextApi);
-    }).asCallback(callback, { spread: !opts.usePostCssInternal });
+          .then(getCSS)
+          .then(processWithTextApi)
+          .then(cleanup);
+
+        function cleanup (result) {
```
Comment (mikelambert) [inline src/uncss.js]: Shouldn't this cleanup callback be on line 218, before the attempts to reference it? Or does it work properly down below, too?

Comment (RyanZim) [reply src/uncss.js]: `function XXX` declarations are hoisted, along with their values, so this works great. You can't do the same thing if you're doing `const xxx = function () { ... }`.

Comment (mikelambert) [reply src/uncss.js]: Ahh interesting. I have a small preference for putting the cleanup function earlier, but other than that, these changes look good to me! (Sorry, but no Windows boxes available, so can't help with that... :/ )

**Comment by RyanZim** [comment]
Fixed @mikelambert's suggestion, and the CI passed. Blame appveyor weirdness, I guess.

**Comment by RyanZim** [comment]
Any objections to merging this?

**Comment by mikelambert** [comment]
No objections

**Pull request merged**